### PR TITLE
Added proposal for JS logging

### PR DESCRIPTION
**Pull request by 0xsarvesh**
Proposal for JS logging

### CODE REVIEW

**Comment by gulshanvasnani** [comment]
I am reviewing this.

File: README.md [hunk 1]
```diff
@@ -90,6 +90,34 @@ Before merging the latest changes from `develop` into a release branch:
 * All build artifacts of a library project should be in the directory `./lib`
 * All documentation should be below the directory `./docs`
 
+### Logging 
+
+* [Winston](https://github.com/winstonjs/winston) should be used as logging library.
+* By default logging should be enabled on the console.
```
Comment (gulshanvasnani) [inline README.md]: `By default logging should be enabled on the console` ==> `By default logging should be enabled to the console`

File: README.md [hunk 2]
```diff
@@ -90,6 +90,34 @@ Before merging the latest changes from `develop` into a release branch:
...
+    * warn
+    * info
+    * verbose
+    * debug
+    * silly
+
+* Default log level should be `info`.  
+* It should be possible to change log level by updating an environment variable.   
+```bash
+export LOG_LEVEL=error
```
Comment (gulshanvasnani) [inline README.md]: `LOG_LEVEL` ==> `ENABLE_PROJECT_NAME_LOG_LEVEL`

**Review by gulshanvasnani — COMMENTED** [review]
Good 👍 
Couple of inline comments.

**Review by gulshanvasnani — APPROVED** [review]
LGTM 👍

File: README.md [hunk 3]
```diff
@@ -90,6 +90,34 @@ Before merging the latest changes from `develop` into a release branch:
 * All build artifacts of a library project should be in the directory `./lib`
 * All documentation should be below the directory `./docs`
 
+### Logging 
+
+* [Winston](https://github.com/winstonjs/winston) should be used as logging library.
+* By default logging to console should be enabled .
```
Comment (schemar) [inline README.md:96]:
```markdown
* By default logging to the console should be enabled.
```

Do we really want to enable logging *by default?* Wouldn't it be weird, if one of your dependencies would start logging to the console?

File: README.md [hunk 4]
```diff
@@ -90,6 +90,34 @@ Before merging the latest changes from `develop` into a release branch:
...
+* It should be possible to switch from console to file logging by updating an environment variable. 
+```bash
+export ENABLE_PROJECT_NAME_LOGGER=true
```
Comment (schemar) [inline README.md:99]: Could also be `PROJECT_NAME_ENABLE_LOGGER`, then all ENVs could start with the project name.

File: README.md [hunk 5]
```diff
@@ -90,6 +90,34 @@ Before merging the latest changes from `develop` into a release branch:
...
+    * warn
+    * info
+    * verbose
+    * debug
+    * silly
+
+* Default log level should be `info`.  
+* It should be possible to change log level by updating an environment variable.   
+```bash
+export ENABLE_PROJECT_NAME_LOG_LEVEL=error
```
Comment (schemar) [inline README.md:113]: Why not simply `PROJECT_NAME_LOG_LEVEL=error`? Why `ENABLE`?

**Review by schemar — COMMENTED** [review]
Should we start extracting that logic into some shared code? 🤔

File: README.md (hunk 3 above)
Comment (0xsarvesh) [reply README.md:96]: If the log level is updated to `error` then the console will only show error, which is normal behavior from most of the libraries.  This also means logger is by default enable.

Thoughts?

File: README.md (hunk 5 above)
Comment (0xsarvesh) [reply README.md:113]: True 👍

File: README.md (hunk 4 above)
Comment (0xsarvesh) [reply README.md:99]: 👍

File: README.md (hunk 3 above)
Comment (0xsarvesh) [reply README.md:96]: Should I make the default log level to `error`?

Comment (schemar) [reply README.md:96]: Hm, I don't know what the best-practice is actually. Someone else?
Maybe @jasonklein @pgev @benjaminbollen @deepesh-kn ?

Comment (jasonklein) [reply README.md:96]: My recollection of when this question came up was that logging would be disabled by default. However, I cannot speak to what is best practice.

@schemar: what are your concerns with dependencies logging to the console / logging enabled by default?

cc: @sarvesh-ost

Comment (schemar) [reply README.md:96]: Just that it is unexpected behavior. We don't know how the consumer uses logs/logging.

**Pull request merged**